### PR TITLE
[MNT] Dataset file restructure

### PR DESCRIPTION
**Pull request by hazrulakmal**
#### Reference Issues/PRs

towards #5105

#### What does this implement/fix? Explain your changes.

This PR aims to enhance the organisation of the datasets module. The current module is disorganized, containing various functions within a single file, leading to a lengthy and unreadable script.

To address this issue, I propose restructuring the module by introducing a new directory/private module called `_readers_writers`. This module will store utility functions responsible for reading and writing data in various time series file formats.

Within this directory, functions will be categorized by format (e.g., 'ts,' 'tsf,' 'arff' etc ). This approach slightly differs from the original suggestion of keeping reader and writer functions separate. I've chosen this approach to avoid redundancy in term of python scripts created and promote a cleaner and more organized codebase.

The test script has also been organised to align with this new style, where all tests for readers and writers are consolidated within a single test script.

Additionally, I've included two utility functions in the API documentation, which were inadvertently omitted previously

The public interface remains the same. 

FYI @achieveordie

### CODE REVIEW

**Comment by hazrulakmal** [comment]
@fkiraly @benHeid @yarnabrina just in case if this is out of radar.

File: sktime/datasets/_readers_writers/tsf.py [hunk 1]
```diff
@@ -0,0 +1,302 @@
+"""Functions for reading time series data from .tsf files."""
```
Comment (fkiraly) [inline sktime/datasets/_readers_writers/tsf.py:1]: typo?

**Review by fkiraly — CHANGES_REQUESTED** [review]
Thanks, much needed reorder!

One request:  it looks like `ts` and `tsf.py` are copy-paste errors? They are duplicated. Can you check?

Non-blocking, I think there might be some docstring cleaning in order, too. I.e., docstrings should optimally say what exacly they assume, which files in which format, etc. And there is not "the standard format" as alleged in some docstrings. But it's perhaps out of scope to fix this - although might not be too much work while you are already there.

File: sktime/datasets/_readers_writers/tsf.py (hunk 1 above)
Comment (hazrulakmal) [reply sktime/datasets/_readers_writers/tsf.py:1]: They are not. they are not duplicates. could you pinpoint which parts in these two files are duplicates?

`tsf` is the file format invented by the people that manage Monash Time Series Forecasting Repository while `ts` is the one that was invented in sktime.

**Comment by hazrulakmal** [comment]
>  I think there might be some docstring cleaning in order, too. I.e., docstrings should optimally say what exactly they assume, which files in which format, etc.

can you clarify which docstring are you referring to? is it the one in the `__init__.py` file or the docstring of the reader and writer util functions?

> One request: it looks like ts and tsf.py are copy-paste errors? They are duplicated. Can you check?

Please refer reply in the review section above.

File: sktime/datasets/_readers_writers/tsf.py (hunk 1 above)
Comment (fkiraly) [reply sktime/datasets/_readers_writers/tsf.py:1]: apologies, this was an inspection error.

**Review by fkiraly — APPROVED** [review]
Apologies, I was mistaken in my comment about `ts` and `tsf`, this was reviewer error.

With that, there is no longer a blocking request.

**Comment by fkiraly** [comment]
Re docstrings (non-blocking), I think there is imprecision in all function docstrings, examples:

```
    DataFrame, ndarray
        If return_separate_X_and_y then a tuple containing a DataFrame and a
        numpy array containing the relevant time-series and corresponding
        class values.
```
should be
```
    (pd.DataFrame, np.ndarray), if `return_separate_X_and_y==True`
        data loaded from the arff file in sktime in-memory representation.
        the `DataFrame` is in `sktime` `pd.DataFrame` format, the `np.ndarray`
        in `numpy1D` format.
```

or in `write_results_to_uea_format`, you have
`Write the predictions for an experiment in the standard format used by sktime.`

I assume this should be "to `ts` format?" though I don´t actually know.

**Pull request merged**